source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -169,6 +170,9 @@ bool FaultFilter::maybeSetupDelay(const Http::RequestHeaderMap& request_headers)
     delay_timer_->enableTimer(duration.value(), &decoder_callbacks_->scope());
     recordDelaysInjectedStats();
     decoder_callbacks_->streamInfo().setResponseFlag(StreamInfo::ResponseFlag::DelayInjected);
+    auto& dynamic_metadata = decoder_callbacks_->streamInfo().dynamicMetadata();
+    (*dynamic_metadata.mutable_filter_metadata())[decoder_callbacks_->filterConfigName()] =
+        fault_settings_->filterMetadata();
```

Comment:
With my exp, the protobuf structure is not fast as we expected. So, I think it would be better to add a simple flag to tell if a meaninigful `filter_metadata` is configured by the users. And then we can only do this injection when the `filter_metadata` is configured explicitly.

For example:


```
// Store the metadata by:
const absl::optional<Envoy::ProtobufWkt::Struct> filter_metadata_;

// Return the metadata by:
OptRef<const Envoy::ProtobufWkt::Struct> filterMetadata() const { // ... }
```

Reply:
sounds reasonable. this is only occurs when delay or fault is injected.